### DISCUSSION TITLE
Connectivity between Sofistik and IDEA StatiCa

### DISCUSSION THREAD

**Discussion by MSSR1521** [idea-statica/ideastatica-public#172] [Help Q&A] [answered]
I am attempting to establish interaction between Sofistik and IDEA StatiCa. As Sofistik is not one of the softwares listed under the BIM links it is my understanding that there is no direct plugin for Sofistik for the export to IDEA StatiCa. I have, however, browsed the tools further, and as far as I can tell, I should be able to use the OpenModel to create a plugin for Sofistik, allowing for an easy link between the two programs. 

My questions are thus: 
1. Please confirm that what I am attempting is possible, and that I am on the right track using IOM. 
2. Assuming the method above is reasonable: As I have no prior experience with IDEA StatiCa, C#, Visual Basics, etc. I am in need of a resource which can walk me through the process of creating the link between Sofistik (a non-BIM link program) and IDEA StatiCa. Are any such resources available? I have been unable to find any, but that may be because I am unsure exactly what keywords to look for.
3. Looking for examples to guide me, I found: https://github.com/idea-statica/iom-examples/blob/master/python-examples/parameters-anchors.py Could this example be used to circumvent the use of C#, as I have a bit more experience with Python?

 
Context for the requests: In Sofistik I have an extensive model using shell elements. I would like to export connections between walls and slabs, and do the detailed modelling of these using IDEA Statica. It is not really an option to manually extract the geometry and loading for each connection, every time the model is run, and as such I wish to automate the process.

**Comment by vojtech-chalupa** [comment]
Hello, thank you for your questions! 
What you are attempting to do is certainly possible. Could you please upload an example of the resulting IDEA StatiCa model which you would like to get automatically from SOFiSTiK?

**Reply by MSSR1521** [reply]
I am hoping to achieve a geometry, roughly as seen below: (I couldn't upload the model-file so here is a screenshot of the geometry)
![Wall-SlabConn](https://user-images.githubusercontent.com/108343383/176195233-56fce3bc-cb2d-4d41-9b03-310b225ebaca.png)
.
Mind you, this is a rudimentary model as I am only just figuring out the possibilities of IDEA StatiCa. The general gist is to model a wall and slab connection. I am open to suggestions.
At some places I may have to place another wall on top.
From Sofistik I intend to extract material data for concrete, thicknesses for wall and slab, section forces for both wall and slab, and any external forces of the connection.
I don't, at the moment, need to extract information about reinforcement, as I am not designing this in Sofistik. So I would be content to manage reinforcement design in IDEA StatiCa's UI.

**Comment by vojtech-chalupa** [comment, marked as answer]
Thank you very much for clarifying. I'm sorry to disappoint, but only IDEA Connection and IDEA RCS projects can be created using IOM and API. We are planning to expand this to IDEA Detail eventually.
